sql/sql_acl.cc

```diff
@@ -11166,6 +11166,21 @@ acl_authenticate(THD *thd, uint com_change_user_pkt_len)
       general_log_print(thd, command, (char*) "%s@%s on %s",
                         mpvio.auth_info.user_name, mpvio.auth_info.host_or_ip,
                         mpvio.db.str ? mpvio.db.str : (char*) "");
+
+    if (log_legacy_user && opt_legacy_user_identifier &&
+        strstr(mpvio.auth_info.user_name, opt_legacy_user_identifier))
+    {
+      /* Log the legacy user */
+      char buffer[256];
+      snprintf(buffer, 256, "LEGACY_USER: %s@%s on %s",
+               mpvio.auth_info.user_name,
+               mpvio.auth_info.host_or_ip,
+               mpvio.db.str ? mpvio.db.str : (char*) "");
+      bool save_enable_slow_log = thd->enable_slow_log;
+      thd->enable_slow_log = true;
```

Comment:
We are writing to the slow log even if the user doesn't have the slow log enabled?  That seems odd.  Is the slow log guaranteed to have a valid file name?